### PR TITLE
PARQUET-955: Only look for Arrow pkgconfig if ARROW_HOME is not set in environment

### PR DESCRIPTION
**Pull request by wesm**
cc @kou 

Ran into this when testing out https://github.com/apache/arrow/pull/548

### CODE REVIEW

File: cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -357,7 +357,14 @@ if(PARQUET_BUILD_BENCHMARKS AND NOT IGNORE_OPTIONAL_PACKAGES)
 endif()
 
 ## Apache Arrow
-pkg_check_modules(ARROW arrow)
+if ("$ENV{ARROW_HOME}" STREQUAL "")
+  # PARQUET-955. If the user has set $ARROW_HOME in the environment, we respect
+  # this, otherwise try to locate the pkgconfig in the system environment
+  pkg_check_modules(ARROW arrow)
```
Comment (majetideepak) [inline cmake_modules/ThirdpartyToolchain.cmake]: Not related to this bug, but should this be moved to `FindArrow.cmake`?

Comment (wesm) [inline cmake_modules/ThirdpartyToolchain.cmake]: Good question for @kou, I am not sure

**Comment by kou** [comment]
I think that it's better that users should use `PKG_CONFIG_PATH` instead of `ARROW_HOME` when they can use it. (For example, Windows users will use `ARROW_HOME` instead of `PKG_CONFIG_PATH`.) Because `PKG_CONFIG_PATH` is more general way than `ARROW_HOME`.

But `ARROW_HOME` is respected than `PKG_CONFIG_PATH` is reasonable. Because `ARROW_HOME` is more specific than `PKG_CONFIG_PATH`. So +1.

File: cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (kou) [inline cmake_modules/ThirdpartyToolchain.cmake]: Yes. We should move to `FindArrow.cmake`.

**Comment by majetideepak** [comment]
I ran into this today. The build system did not honor `ARROW_HOME`  and was picking old arrow headers causing a very weird segmentation fault. I would like to print the include path along with the library path that gets printed. I would also like move this to `FindArrow.cmake`. I can open a separate JIRA to do this if it is not in the scope of this JIRA.

**Comment by wesm** [comment]
Just did both, could you both take a look and see if things look reasonable?

File: cmake_modules/FindArrow.cmake [hunk 2]
```diff
@@ -18,52 +18,82 @@
...
-  ${ARROW_SEARCH_HEADER_PATHS}
-  # make sure we don't accidentally pick up a different version
-  NO_DEFAULT_PATH
-)
+  set(ARROW_LIB_NAME ${CMAKE_SHARED_LIBRARY_PREFIX}arrow)
+
+  set(ARROW_SHARED_LIB ${ARROW_LIBDIR}/${ARROW_LIB_NAME}${CMAKE_SHARED_LIBRARY_SUFFIX}${ARROW_SHARED_LIB_SUFFIX})
+  set(ARROW_STATIC_LIB ${ARROW_LIBDIR}/${ARROW_LIB_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX})
+else()
+  if( NOT "$ENV{ARROW_HOME}" STREQUAL "")
```
Comment (majetideepak) [inline cmake_modules/FindArrow.cmake]: Do we need this check again here? if `ARROW_HOME` is set in the env, `ARROW_FOUND` is always false right?

Comment (wesm) [inline cmake_modules/FindArrow.cmake]: I moved the `set(ARROW_HOME ...)` to the top

**Comment by majetideepak** [comment]
+1 LGTM.  I tried this patch locally and it does seem to honor `ARROW_HOME` if set.
Thanks @wesm

**Comment by xhochy** [comment]
@wesm Can you also update this for `pyarrow` so that the `FindArrow` scripts don't get out-of-sync (which they probably are already)?

**Comment by wesm** [comment]
yep

**Pull request closed without merging**